boa/src/builtins/value/mod.rs

```diff
@@ -723,16 +723,33 @@ impl ValueData {
     /// Conversts the `Value` to `JSON`.
     pub fn to_json(&self) -> JSONValue {
         match *self {
-            Self::Null | Self::Symbol(_) | Self::Undefined => JSONValue::Null,
+            Self::Null => JSONValue::Null,
             Self::Boolean(b) => JSONValue::Bool(b),
             Self::Object(ref obj) => {
-                let new_obj = obj
-                    .borrow()
-                    .properties
-                    .iter()
-                    .map(|(k, _)| (k.clone(), self.get_field_slice(k).to_json()))
-                    .collect::<Map<String, JSONValue>>();
-                JSONValue::Object(new_obj)
+                if obj.borrow().kind == ObjectKind::Array {
+                    let mut arr: Vec<JSONValue> = Vec::new();
+                    obj.borrow().properties.iter().for_each(|(k, _)| {
```

Comment:
I think it would make sense to only iterate over `properties.keys()`, since we don't use the values.